### PR TITLE
feat(ootle_wasm): add stealth crypto primitives

### PR DESCRIPTION
**Pull request by martinserts**
Description
---

Adds WASM bindings for the full stealth-transfer crypto surface — output statement generation (with range proofs + viewable-balance proofs), balance proof signing/validation, inbound UTXO decryption, DH key derivation, ElGamal
  view-key decryption, and a transfer validator. Also fixes the pre-existing `wasm-pack` build break by enabling `getrandom`'s `wasm_js` feature for the v0.4 line pulled in transitively.

Motivation and Context
---

The Python Ootle client currently relies on the wallet daemon to perform stealth crypto operations. To enable a daemon-less client, the WASM library needs to expose the same primitives directly. This PR closes the gap and unblocks single-input stealth transfers end-to-end from Python.

How Has This Been Tested?
---

  - 42 unit tests in `ootle-wasm-core` covering round-trips, validation against the engine, error paths, and edge cases (empty arrays, wrong keys, out-of-range values).
  - `wasm-pack build --release` produces a working 1.1 MB WASM package (408 KB gzipped) with all 12 new exports surfaced in the generated TypeScript bindings.

What process can a PR reviewer use to test or verify this change?
---

  1. `cargo test -p ootle-wasm-core` — runs the test suite.
  2. `cd crates/ootle_wasm && ./build.sh bundler release` — produces the npm package; check `pkg/ootle_wasm.d.ts` for the new exports.
  3. Optionally publish a local build and run the Python client's stealth-transfer flow against an Ootle network.

Breaking Changes
---

- [x] None
- [ ] Requires data directory to be deleted
- [ ] Other - Please specify

### CODE REVIEW

File: crates/ootle_wasm/core/src/stealth/types.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        return Err(OotleWasmError::InvalidByteLength {
+            field,
+            expected: N,
+            got: bytes.len(),
+        });
+    }
+    let mut out = [0u8; N];
+    out.copy_from_slice(&bytes);
+    Ok(out)
+}
```
[the remark this answers is not in the document]
Comment (martinserts) [reply crates/ootle_wasm/core/src/stealth/types.rs:141]: @gemini-code-assist Good idea, adjusted - [44428bb](https://github.com/tari-project/tari-ootle/pull/2134/commits/44428bb4f9e7284a1f9fdf1d99d611f014090002)

**Review by sdbondi — APPROVED** [review]
